### ISSUE TITLE
Django Dirtyfields is painfully slow with m2m relations

### ISSUE THREAD

**Issue by opit** [networklocum/django-changes-on-save#13]
```from practice.models import Practice; prs = Practice.objects.all(); print(len(prs))```
takes ~12s for ~2500 practices.

If the definition for Practice is changed from:

```
class Practice(AnnounceChangesMixin, models.Model):
```

to:

```
class Practice(models.Model):
```

Results to above query are near-instant <1s.

Related issue & PR on django-dirtyfields:
https://github.com/romgar/django-dirtyfields/issues/73
https://github.com/romgar/django-dirtyfields/pull/76

This issue will impact the performance of every query, read & write for all our tables that have M2M relationships and use the Announce Mixin.

Possible fix: Patch Django Dirtyfields, do we need the m2m functionality?

**Comment by romgar** [comment]
Hi @opit ! 

I'm adding a parameter named `ENABLED_M2M_CHECK`, disabled by default, to avoid this kind of issue.
It will be released soon in version `1.1` of `django-dirty-fields`.

**Comment by opit** [comment]
@romgar Hero!

`run pip install django-dirtyfields==1.1`
